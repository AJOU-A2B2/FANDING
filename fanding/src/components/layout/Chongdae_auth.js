import React, {Component} from 'react';
import { Card, Button, CardTitle, CardText, Row, Col } from 'reactstrap';
import { Link } from "react-router-dom";

class Chongdae_auth extends Component{

    render()
    {
        return(
          <Card body>
                <CardTitle>본인 인증</CardTitle>
                <CardText>펀딩을 생성하려면 본인 인증이 필요합니다.</CardText>
                <Link to="/account_auth">
                  <Button color="warning" onClick={chongdae_auth} >본인 인증</Button>
                </Link>
          </Card>  
        );
    }
}

function chongdae_auth(){
  const chongdaeVerify = window.open(
    "https://testapi.openbanking.or.kr/oauth/2.0/authorize?"
    +"response_type=code&client_id=qhsl7X3L59LPtU6QfdZNv2d4jYYKKFiY8K2iw2NI&"
    +"redirect_uri=http://localhost:3000/account_auth&"
    +"scope=login inquiry transfer&state=12345678901234567890123456789012&auth_type=0&lang=kor"
  )
};



export default Chongdae_auth;