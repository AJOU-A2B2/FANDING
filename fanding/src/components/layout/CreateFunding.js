import React, {Component} from 'react';
import { Button, Col, Form, FormGroup, Label, Input, FormText, CustomInput } from 'reactstrap';
import { Link } from "react-router-dom";
import {BsStop} from "react-icons/bs"
import { connect } from 'react-redux';
import {firebase_funding_save} from '../../store/actions/formActions';
import {storage} from "../../config/fbConfig";
import { Redirect } from 'react-router-dom';


class CreateFunding extends Component{

    constructor(props) {
        super(props);
        this.state={
            image:null,
            url:"",
            progress:0,
            artistSelect: '',
            fundingType: '',
            fundingTitle: '',
            fundingStartDate: '',
            fundingEndDate: '',
            fundingStartTime: '',
            fundingEndTime: '',
            fundingPeriodLimit: '',
            // thumbnailImage: '',
            detailText: '',
            itemTitle:'',
            itemPrice:'',
            itemLimitBox:'',
            itemRemain:'',
            itemLimit:'',
            shippingMethod:'',
            shippingFee:'',
            shippingDetail:'',
            redirectToReferrer: false
        };
    }

    handleImageChange = e => {

        if (e.target.files[0]) {
            this.setState({image: e.target.files[0]});
        }

    }

    handleChange = e => {
        this.setState({
            [e.target.id]: e.target.value,
        });
    };

    handleSubmit = e => {


        e.preventDefault();
        if(this.state.image!=null){
            const uploadTask = storage.ref(`images/${this.state.image.name}`).put(this.state.image);
            uploadTask.on(
                "state_changed",
                snapshot => {
                    const progress = Math.round(
                        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
                    );
                    this.setState({progress: progress});
                },
                error => {
                    console.log(error);
                },
                () => {
                    storage
                        .ref("images")
                        .child(this.state.image.name)
                        .getDownloadURL()
                        .then(url => {
                            this.setState({url:url});
                            console.log(this.state.url)
                            // console.log(this.state);
                            this.setState({image:this.state.image.name})
                            console.log(this.state);
                            this.props.firebase_funding_save(this.state)
                            this.setState({redirectToReferrer: true})

                        })
                }
            )
        }
        else{
            this.props.firebase_funding_save(this.state);
            this.setState({redirectToReferrer: true})

        }


    };

    handleClick = e =>{
        // history.replace("/");
    }

    render()
    {

        if(this.state.redirectToReferrer===true){
            alert("펀딩이 생성되었습니다.");
           return  <Redirect to='/' />
        }
        return (
            <>
                <Form>
                    <FormGroup>
                        <Label for="artistSelect">아티스트</Label>
                        <CustomInput type="select" id="artistSelect" name="customSelect" onChange={this.handleChange}>
                            <option value="">Select</option>
                            <option>BTS</option>
                            <option>BLACKPINK</option>
                            <option>TWICE</option>
                            <option>ITZY</option>
                            <option>GOT7</option>
                            <option>기타</option>
                        </CustomInput>
                    </FormGroup>

                    <FormGroup>
                        <Label for="Radio">펀딩 유형 선택</Label>
                        <div>
                            <CustomInput type="radio" id="fundingType" name="customRadio" label="리워드형 펀딩"  onChange={this.handleChange} inline/>
                            <CustomInput type="radio" id="fundingType" name="customRadio" label="모금형 펀딩"  onChange={this.handleChange} inline/>
                        </div>
                    </FormGroup>

                    <FormGroup>
                        <Label for="fundingTitle">펀딩 제목</Label>
                        <Input type="text" name="title" id="fundingTitle"
                               placeholder="펀딩 제목을 입력하세요"
                               onChange={this.handleChange}/>
                    </FormGroup>
                </Form>

                <Form inline>
                    <FormGroup>
                        <Label for="startDate">펀딩 기간(입금 기간)</Label>
                        <Input
                            type="date"
                            name="date"
                            id="fundingStartDate"
                            placeholder="펀딩 시작일"
                            onChange={this.handleChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Input
                            type="time"
                            name="time"
                            id="fundingStartTime"
                            placeholder="00:00"
                            onChange={this.handleChange}
                        />
                    </FormGroup>

                    <FormGroup className="ml-auto">
                        <Input
                            type="date"
                            name="date"
                            id="fundingEndDate"
                            placeholder="펀딩 종료일"
                            onChange={this.handleChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Input
                            type="time"
                            name="time"
                            id="fundingEndTime"
                            placeholder="00:00"
                            onChange={this.handleChange}
                        />
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleCheckbox"></Label>
                        <CustomInput type="checkbox" id="fundingPeriodLimit" label="기간 제한 없음" onChange={this.handleChange}/>
                    </FormGroup>
                </Form>

                <Form>
                    <FormGroup>
                        <Label for="fileBrowser">썸네일 이미지 (size: 350*250)</Label>
                        <CustomInput type="file" id="thumbnailImage" name="customFile"
                                     label="이미지를 업로드 하세요"
                                     onChange={this.handleImageChange}
                        />


                    </FormGroup>

                    <FormGroup>
                        <Label for="detailText">상세 설명</Label>
                        <Input type="textarea" name="text" id="detailText" onChange={this.handleChange}/>
                    </FormGroup>
                </Form>

                <Label for="itemInfo">상품정보</Label>
                <Form inline>

                    <FormGroup>
                        <Label for="itemTitle">상품명</Label>
                        <Input type="text" name="title" id="itemTitle" placeholder="" onChange={this.handleChange}/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="itemPrice">가격</Label>
                        <Input type="text" name="title" id="itemPrice" placeholder="" onChange={this.handleChange}/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="itemRemain">재고</Label>
                        <Input type="text" name="title" id="itemRemain" placeholder="" onChange={this.handleChange}/>
                    </FormGroup>

                    <FormGroup>
                        <CustomInput type="checkbox" id="itemLimitBox" label="구매개수 제한" onChange={this.handleChange} />
                        <Input type="text" name="title" id="itemLimit" placeholder="" onChange={this.handleChange}/>
                    </FormGroup>
                    {/*상품추가 버튼 만들기*/}
                </Form>

                <Label for="shipping">배송</Label>
                <Form inline>
                    <FormGroup>
                        <Label for="shippingMethod">배송방법</Label>
                        <Input type="text" name="title" id="shippingMethod"
                               placeholder="예)택배, 준등기"
                               onChange={this.handleChange}
                        />
                    </FormGroup>
                    <FormGroup className="ml-3">
                        <Label for="shippingFee">배송비</Label>
                        <Input type="text" name="title" id="shippingFee"
                               placeholder="0"
                               onChange={this.handleChange}
                        />
                    </FormGroup>
                </Form>

                <Form>
                    <Label for="shippingDetail">배송 안내</Label>
                    <Input type="textarea" name="text" id="shippingDetail"
                           onChange={this.handleChange}
                    />
                </Form>
                {/*배송방법추가 버튼 만들기*/}

                <Form onSubmit={this.handleSubmit}>
                    {/*<Link to='/'>*/}

                    {/*    <Button color="warning" size="lg" block onChange={this.handleClick}>폼 만들기</Button>*/}
                    {/*</Link>*/}
                    <Button color="warning" size="lg" block >폼 만들기</Button>
                </Form>
            </>
        )

    }
}


const mapStateToProps = (state) => {
    return{
        authError: state.auth.authError,
        auth: state.firebase.auth
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        firebase_funding_save: (creds) => dispatch(firebase_funding_save(creds))
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(CreateFunding);