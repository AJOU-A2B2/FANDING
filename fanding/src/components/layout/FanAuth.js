import React, {Component} from 'react';
import {Container, Form, FormGroup, Label, CustomInput, Media, Button} from 'reactstrap';
import ticketImg from './ticket.png';
import streamingImg from './streaming.png';
import fanclub from './fanclub.png';
import {storage} from "../../config/fbConfig";
import firebase from "firebase";

const imgStyle = {
    maxHeight: 400,
    maxWidth: 200
  }

class FanAuth extends Component{

    constructor(props) {
        super(props);
        this.state={
            streamingImage:"",
            fanclubImage:"",
            concertImage:"",
            albumImage:"",
            images:[],
            urls:[]
        };
    }


    handleImageChange = e => {


        var random=Math.random();
        var num=Math.floor(random*10000000000);
        var name=e.target.files[0].name.split(".");

        const image_id=e.target.id;
        const image_name=name[0]+num+"."+name[1];
        const image_file=e.target.files[0]

        this.setState({
            images:this.state.images.concat({id:e.target.id,image:e.target.files[0]})
        },()=>{

            const uploadTask = storage.ref(`fan_auth/${image_name}`).put(image_file);
            uploadTask.on(
                "state_changed",
                snapshot => {
                    //             // const progress = Math.round(
                    //             //     (snapshot.bytesTransferred / snapshot.totalBytes) * 100
                    //             // );
                    //             // this.setState({progress: progress});
                },
                error => {
                    console.log(error);
                },
                () => {
                    storage
                        .ref("fan_auth")
                        .child(image_name)
                        .getDownloadURL()
                        .then(url => {
                            // this.props.nextPage();
                            this.setState({
                                urls:this.state.urls.concat({id:image_id,url:url})
                            });
                        })
                }
            )
        });

        // console.log(this.state)

    }

    handleSubmit=(e)=>{
        // console.log(this.state.images.length);
        // console.log(this.state.urls.length);


        if(this.state.images.length!==0){
            // console.log(this.state.urls)
            if(this.state.images.length===this.state.urls.length){
                this.props.handler(this.state.urls);

            }
        }
        else{
            alert("최소 하나의 팬 인증 사진을 첨부하세요")
        }

    }
    render()
    {
        return(
            <>
            <section className="gallery5 mbr-gallery cid-sgtDmxvlJH" id="gallery5-q">
            <div className="mbr-section-head" style={{paddingBottom: '30px'}}>
                       <h3 className="mbr-section-title mbr-fonts-style align-center m-0 pb-30 mb-10 display-2">
                           <strong>팬 인증하기</strong></h3>
                       
           </div>
           <Container
           style={{backgroundColor:"#fafafa", borderRadius:"10px", padding:"3em 2em"}}>
            <p>팬 인증은 <strong>스트리밍 인증, 팬클럽 인증, 공연 관람 내역 중 최소 하나</strong>를 하셔야합니다.</p>
            <p>팬 인증은 펀딩 참여자가 펀딩을 고르는 하나의 기준이 될 수 있습니다.</p>
            <p>모든 사진에는 도용 방지를 위해 <strong>자신의 닉네임, 날짜, 시간</strong>이 나올 수 있게 해주세요.</p>
            <Form>
                <FormGroup>
                <Label><strong>1. 스트리밍 인증하기</strong></Label>
                <Media object src={streamingImg} style={imgStyle}/>
                <CustomInput type="file" id="streamingImage" name="customFile"
                                     label="스트리밍 인증 캡처를 업로드 하세요"
                                     onChange={this.handleImageChange}
                        />
                </FormGroup>
            
                <FormGroup>
                <Label><strong>2. 팬클럽 인증하기</strong></Label>
                <Media object src={fanclub} style={imgStyle}/>
                <CustomInput type="file" id="fanclubImage" name="customFile"
                                     label="팬클럽 인증 캡처를 업로드 하세요"
                                     onChange={this.handleImageChange}
                        />
                </FormGroup>
            
                <FormGroup>
                <Label><strong>3. 공연 관람 내역 인증하기</strong></Label>
                <Media object src={ticketImg} style={imgStyle}/>
                <CustomInput type="file" id="concertImage" name="customFile"
                                     label="공연 관람 내역을 업로드 하세요"
                                     onChange={this.handleImageChange}
                        />
                </FormGroup>
            
                <FormGroup>
                <Label><strong>4. 앨범 및 공식 굿즈 인증하기</strong></Label>
                <CustomInput type="file" id="albumImage" name="customFile"
                                     label="앨범 및 공식 굿즈를 업로드 하세요"
                                     onChange={this.handleImageChange}
                        />
                </FormGroup>
                <Button color="warning" size="md" onClick={this.handleSubmit} block >다음</Button>
                {/* 버튼 handler 연결해야함 */}
            </Form>

         </Container>
         </section>
         </>   
        )
    }
}

export default FanAuth;