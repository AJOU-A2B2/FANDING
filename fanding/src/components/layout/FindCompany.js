import React, {Component} from 'react';
import { Link } from "react-router-dom";
import { Button, NavLink, Col, Form, FormGroup, Label, Input, 
    Card, CardBody, CardTitle, CardSubtitle, CardImg, CardText,
    FormText, CustomInput } from 'reactstrap';
import { connect } from 'react-redux';

const FindCompany = (props) => {

    return (
        <div>
        <NavLink href="/find_company_form"><Button className="pull-right" outline color="primary">업체모집 폼 생성</Button></NavLink>

        
        <Card sm="6">
            <CardImg top width="100%" src="/assets/318x180.svg" alt="Card image cap" />
            <CardBody>
            <CardTitle>Fuding title1</CardTitle>
            <CardSubtitle> subtitle</CardSubtitle>
            <CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>
            {/*<Button>Button</Button>*/}
            </CardBody>
        </Card>
        <Card sm="6">
            <CardImg top width="100%" src="/assets/318x180.svg" alt="Card image cap" />
            <CardBody>
            <CardTitle>Fuding title2</CardTitle>
            <CardSubtitle> subtitle</CardSubtitle>
            <CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>
            {/*<Button>Button</Button>*/}
            </CardBody>
        </Card>
        <Card sm="6">
            <CardImg top width="100%" src="/assets/318x180.svg" alt="Card image cap" />
            <CardBody>
            <CardTitle>Fuding title3</CardTitle>
            <CardSubtitle> subtitle</CardSubtitle>
            <CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>
            {/*<Button>Button</Button>*/}
            </CardBody>
        </Card>
        </div>
    )    
    
}

export default FindCompany; 