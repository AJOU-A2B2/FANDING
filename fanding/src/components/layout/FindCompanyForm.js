import React, {Component} from 'react';
import { Link } from "react-router-dom";
import { Button, NavLink, Col, Form, FormGroup, Label, Input, FormText, CustomInput } from 'reactstrap';
import { connect } from 'react-redux';
import {firebase_recruit_save} from '../../store/actions/recruitCompanyActions';
import {storage} from "../../config/fbConfig";
import { Redirect } from 'react-router-dom';
//toast-ui
import { Editor } from '@toast-ui/react-editor';
import 'tui-color-picker/dist/tui-color-picker.css';
import 'codemirror/lib/codemirror.css';
import '@toast-ui/editor/dist/toastui-editor.css';
import colorSyntaxPlugin from '@toast-ui/editor-plugin-color-syntax';
import hljs from "highlight.js";
import codeSyntaxHighlightPlugin from '@toast-ui/editor-plugin-code-syntax-highlight';
//chart plugin
import 'tui-chart/dist/tui-chart.css';
import chart from '@toast-ui/editor-plugin-chart';

class FindCompanyForm extends Component {

    constructor(props){
        super(props);

        this.state={
            image:null,
            url:"",
            progress:0,
            itemTitle: '',
            itemImage:'',
            detailText:'',
            itemPrice:'',
            itemRemain:'',
            shippingMethod:'',
            content:''
        };
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    

    editorRef = React.createRef();

    handleImageChange = e => {

        if (e.target.files[0]) {
            this.setState({image: e.target.files[0]});
        }

    }

    handleChange = e => {
        this.setState({
          [e.target.id]: e.target.value,
        });
      };

   
    
    handleChangeEditor = e =>
    {
        const content = this.editorRef.current.getInstance().getHtml();
        console.log("I am editor" + content)

        this.setState({
            content: content
        });
    }
    handleSubmit = e => {
        e.preventDefault();
        
        console.log(this.state.content);

        if(this.state.image!=null){
            const uploadTask = storage.ref(`images/${this.state.image.name}`).put(this.state.image);
            uploadTask.on(
                "state_changed",
                snapshot => {
                    const progress = Math.round(
                        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
                    );
                    this.setState({progress: progress});
                },
                error => {
                    console.log(error);
                },
                () => {
                    storage
                        .ref("images")
                        .child(this.state.image.name)
                        .getDownloadURL()
                        .then(url => {
                            this.setState({url:url});
                            console.log(this.state.url)
                            // console.log(this.state);
                            this.setState({image:this.state.image.name})
                            console.log(this.state);
                            this.props.firebase_recruit_save(this.state)
                            this.setState({redirectToReferrer: true})

                        })
                }
            )
        }
        else{
            this.props.firebase_recruit_save(this.state);
            this.setState({redirectToReferrer: true})

        }
    };
    render()
    {
        //const { authError, auth } = this.props;
        //if (auth.uid) return <Redirect to='/' />
        if(this.state.redirectToReferrer===true){
            alert("업체 모집폼이 생성되었습니다");
           return  <Redirect to='/' />
        }
        return (
            <>
            <Form>
                <FormGroup>
                    <Label for="itemTitle">제작할 상품 종류</Label>
                    <Input type="text" name="title" id="itemTitle" 
                    placeholder="제작할 상품 종류 ex)키링, 그립톡"
                    onChange={this.handleChange}/>
                </FormGroup>
            </Form>
                
                <Form>
                <FormGroup>
                    <Label for="fileBrowser">예상 디자인 (size: 350*250)</Label>
                    <CustomInput type="file" id="itemImage" name="customFile" 
                    label="이미지를 업로드 하세요" 
                    onChange={this.handleImageChange}
                    />
                </FormGroup>
                </Form>

                <Form>
                <FormGroup>
                    <Label for="detailText">상세 설명</Label>
                    <Editor
                        previewStyle="vertical"
                        height="400px"
                        initialEditType="wysiwyg"
                        initialValue="hello"
                        ref={this.editorRef}
                        plugins= {[codeSyntaxHighlightPlugin.bind(hljs), colorSyntaxPlugin, chart]}
                        onChange = {this.handleChangeEditor}
                        />
                

                        <div id="toastEditor">
                            <h1>Toast UI Editor Example</h1>
                            <div id="editSection"></div>
                            {/*<button onClick={this.saveArticle} className="btn_save">Save</button>*/}
                            <button onChange={this.handleChangeEditor} className="btn_save">Save</button>
                            <div>
                                <h2>result</h2>
                                <textarea className="tf_result" value={this.state.content} readOnly="readOnly"></textarea>
                            </div>
                        </div>
                </FormGroup>
                </Form>
                
                <Form inline>
                <FormGroup>
                    <Label for="itemPrice">원하는 가격</Label>
                    <Input type="text" name="title" id="itemPrice" placeholder="" onChange={this.handleChange}/>
                </FormGroup>
                <FormGroup>
                    <Label for="itemRemain">예상 제작 개수</Label>
                    <Input type="text" name="title" id="itemRemain" placeholder="" onChange={this.handleChange}/>
                </FormGroup>
                </Form>
                
                <Label for="shipping">배송</Label>
                <Form>
                    <FormGroup>
                    <Label for="shippingMethod">원하는 배송방법</Label>
                    <Input type="text" name="title" id="shippingMethod" 
                    placeholder="예)택배, 준등기"
                    onChange={this.handleChange}
                    />
                    </FormGroup>
                </Form>
                
                
                
                <Form onSubmit={this.handleSubmit}>
                    {/*<Link to='/find_company'>

                        <Button color="warning" size="lg" block>폼 만들기</Button>
                    </Link>
                    */}
                <Button color="warning" size="lg" block onChange={this.handleClick}>폼 만들기</Button>
                </Form>            
            </>
        )

    }
}

const mapStateToProps = (state) => {
    return{
        authError: state.auth.authError,
        auth: state.firebase.auth
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        firebase_recruit_save: (creds) => dispatch(firebase_recruit_save(creds))
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(FindCompanyForm);