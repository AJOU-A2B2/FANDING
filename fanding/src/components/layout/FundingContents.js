import React from 'react';
import { TabContent, TabPane, Nav, NavItem, NavLink, Card, Button, CardTitle, CardText, Row, Col, CardImg, CardBody,
    CardSubtitle } from 'reactstrap';
import { Link } from 'react-router-dom';

const FundingContents = ({funding})=>{


    // if(funding.createTime){
    //     var date= new Date(funding.createTime.seconds*1000);
    //     console.log(date)
    //     console.log(funding.thumbnailImage)
    //
    // }
    // const data= find(funding);

    // console.log("Funding Contents:",funding);
    return(

        <div>
            <Link to={'funding_detail/'+funding.id} funding={funding}>
            <Card sm="6">
                <CardImg src={funding.thumbnailImage}top width="100" alt="Card image cap" />
                <CardBody>
                    <CardTitle>[{funding.artistSelect}] {funding.fundingTitle}</CardTitle>
                    <CardSubtitle> {funding.fundingStartDate} {funding.fundingStartTime} ~ {funding.fundingEndTime} {funding.fundingEndTime}</CardSubtitle>
                    {/*<CardText>Some quick example text to build on the card title and make up the bulk of the card's content.</CardText>*/}
                    {/*<Button>Button</Button>*/}
                </CardBody>
            </Card>
            </Link>
        </div>
    )
}

export default FundingContents;

