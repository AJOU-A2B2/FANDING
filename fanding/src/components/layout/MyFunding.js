import React, {Component} from 'react';
import {connect} from 'react-redux';
import {loadFundings} from "../../store/actions/userActions";
import FundingList from "./FundingList";
import {Container, Row, Col, CardDeck} from "reactstrap";
import SideBar from "./SideBar"
import TopNavbar from "./TopNavbar"
class MyFunding extends Component {

    constructor(props) {
        super(props);
        this.state={

        }
    }

    // componentDidMount() {
    //     console.log("didmount")
    //     const uid=this.props.params.uid;
    //     console.log(uid)
    //
    //     this.props.dispatch(loadFundings())
    //     // this.setState({data_state:true})
    // }
    componentWillUnmount(){
        console.log("unmount")
    }


    // componentDidUpdate(prevProps, prevState, snapshot) {
    //     if(this.props.auth !== prevProps.auth){
    //             this.props.dispatch(loadFundings(this.props.auth.uid))
    //
    //     }
    //     }
    componentDidMount() {
        this.props.dispatch(loadFundings(this.props.auth.uid))
    }


    render(){
        console.log("render");

        const {auth,user_data}=this.props;
        console.log(this.props);

        if(this.props.user_data.length!==0){
            // console.log(user_data);
            return(

                <section class="gallery5 mbr-gallery cid-sgtDmxvlJH" id="gallery5-q"
                style={{paddingTop:'25px'}}>
                    <Container>
                        <Row>
                            <Col sm={3}>
                            <TopNavbar />
                            <SideBar />
                            </Col>
                            <Col>

                                    <FundingList fundings={user_data}/>
                            
                            </Col>
                        </Row>
                    </Container>
                </section>
            )
        }
        else{

            return(

                <section class="gallery5 mbr-gallery cid-sgtDmxvlJH" id="gallery5-q"
                style={{paddingTop:'25px'}}>
                    <Container>
                        <Row>
                            <Col sm={3}>
                            <TopNavbar />
                            <SideBar />
                            </Col>
                            <Col>
                            
                    <h2 style={{paddingTop:'90px'}}>생성한 펀딩이 없습니다.</h2>
                    </Col>
                        </Row>
                    </Container>
                </section>

            )
        }


    }


}

// store의 state 값을 props로 연결해줌
const mapStateToProps = (state) => {
    return{
        authError: state.auth.authError,
        auth: state.firebase.auth,
        user_data: state.auth.user_data
    }
};
export default connect(
    mapStateToProps
)(MyFunding);
