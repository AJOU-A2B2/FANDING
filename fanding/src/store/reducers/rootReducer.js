import authReducer from "./authReducer";
import formReducer from "./formReducer";
import paymentReducer from "./paymentReducer";
import verifyReducer from "./chongdaeReducer";
import { combineReducers } from "redux";
import { firestoreReducer } from "redux-firestore";
import { firebaseReducer } from "react-redux-firebase";

const rootReducer = combineReducers({
  auth: authReducer,
  form: formReducer,
  payment: paymentReducer,
  verify: verifyReducer,
  firestore: firestoreReducer,
  firebase: firebaseReducer,
});

export default rootReducer;
