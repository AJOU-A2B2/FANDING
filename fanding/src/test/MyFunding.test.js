import React from 'react';
import renderer from 'react-test-renderer';
import MyFunding from "../components/layout/MyFunding";
import {mapStateToProps} from "../components/layout/MyFunding";
import {describe, it} from "@jest/globals";
import {Provider} from 'react-redux';
import { BrowserRouter, Route, Switch } from "react-router-dom";
import {shallow, configure,mount} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {getFirebase, ReactReduxFirebaseProvider} from "react-redux-firebase";
import firebase from "firebase";
import fbconfig from "../config/fbConfig";
import {createFirestoreInstance, getFirestore, reduxFirestore} from "redux-firestore";
import {jest} from '@jest/globals'
import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../store/reducers/rootReducer";
import thunk from "redux-thunk";
import logger from "redux-logger";

configure({adapter:new Adapter()});

let container;
// let store;
// const mockStore = configureMockStore();
// const initState ={
//     authError: null,
//     auth: null,
//     user_data: []
// }
// store = mockStore(initState)
const store = createStore(rootReducer,
    compose(
        applyMiddleware(thunk.withExtraArgument( { getFirebase, getFirestore }),logger),
        reduxFirestore(fbconfig),
    ) //enhancer
);
const rrfProps = {
    firebase,
    config: fbconfig,
    dispatch: store.dispatch,
    createFirestoreInstance,
};
let props;
props={
    authError: null,
    auth: null,
    user_data: []
}

beforeAll(() => {
    jest.mock('react-redux');
});

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);

});
afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

describe('can render', () => {
    // 첫 render와 componentDidMount를 테스트


    let component;
    let dispatch;
    beforeEach(()=>{

        dispatch = jest.fn().mockReturnValue(Promise.resolve({
            user_data: []
        }));
        component = shallow(
            <Provider store={store}>
                 <ReactReduxFirebaseProvider {...rrfProps}>
                     <BrowserRouter>
                         <MyFunding props={props}/>
                     </BrowserRouter>
                 </ReactReduxFirebaseProvider>
            </Provider>
        )
    });

    it('should check componentDidMount()', () => {


        // expect(dispatch).toBeCalled();
        // const instance=component.instance();
        // const loadFundings=jest.fn()
        // console.log(instance)

        // jest.spyOn(instance,loadFundings());
        // instance.componentDidMount();
        // expect(instance.dispatch(loadFundings())).toHaveBeenCalledTimes(1);
        // const dispatch=jest.fn()
        // if(firebase.auth().currentUser){
        //     expect(dispatch(loadFundings(firebase.auth().currentUser.uid)));
        //
        // }
        // else{
        //     expect(props.user_data=[])
        // }
    });

    it('matches snapshot', () => {
        // const result=renderer.create(component)
        // const tree=result.toJSON();
        // expect(tree).toMatchSnapshot();
    });

    it('renders fundings', () => {
       // const text=component.find('h2');
       // console.log(text);
       // expect(text.toEqual(['생성한 펀딩이 없습니다.']))

        // const container = component.find("gallery5 mbr-gallery cid-sgtDmxvlJH");
        // console.log(container.debug())
        // expect(component.find('section'))

    });

});