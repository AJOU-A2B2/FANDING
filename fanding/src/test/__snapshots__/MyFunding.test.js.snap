// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render matches snapshot 1`] = `ShallowWrapper {}`;
